src/Compilers/Core/Portable/Collections/TemporaryArrayExtensions.cs

```diff
@@ -121,6 +121,20 @@ private static void ThrowSequenceContainsMoreThanOneElement()
             return default;
         }
 
+        public static int IndexOf<T, TArg>(this in TemporaryArray<T> array, Func<T, TArg, bool> predicate, TArg arg)
```

Comment:
I'm going to continue to object to new methods being introduced that put `TArg` last. Yes, I understand that we have a general pattern in Roslyn of putting them last. This pattern is bad, as it forces us to either go fill in the last parameter first, or to write the body of the predicate lambda without any IDE features on the TArg parameter because the IDE doesn't know what it is yet.

Reply:
I'm not against running a refactoring that swaps these arguments for all methods we have.

Reply:
Fwiw, I would *very* much like to swap these (in a future pr).  It's very annoying to hang to write: 

`callback: null, (... Figure out args...)` them hop back and forth to flesh this out.  

It also reads much worse imo with the large lambda coming first without a sense of the small list of values that are captured into it and passed along. 

That said, let's fix this later.